### PR TITLE
Test dma driver on some stm32 target boards

### PR DESCRIPTION
**Pull request by FRASTM**
Enable the dma DTS to run the `tests/drivers/dma/loop_transfer` and `tests/drivers/dma/chan_blen_transfer` on the following boards:

- nucleo_f091rc
- nucleo_l152re
- nucleo_f429zi
- nucleo_f103rb
- nucleo_f207zg

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
- adding the dma feature in the yaml and updating the documentation
- removing stm32f3_disco since the #37162
- removing nucleo_ l103rb: no hardware target defined

**Comment by FRASTM** [comment]
For the tests/drivers/dma on the **nucleo_l152rre** target board, the DTS is updated to add the definitio of the dma1 instance (commit )

**Comment by FRASTM** [comment]
Going to enable the dma in each board DTS  instead of an overlay.

File: boards/arm/nucleo_f429zi/nucleo_f429zi.yaml [hunk 1]
```diff
@@ -21,4 +21,5 @@ supported:
   - usb_device
   - watchdog
   - adc
-  - dac
+  - adc
```
Comment (erwango) [inline boards/arm/nucleo_f429zi/nucleo_f429zi.yaml:24]: "dac" renamed as "adc"

File: tests/drivers/dma/chan_blen_transfer/boards/nucleo_l152re.overlay [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (c) 2021 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&dma1 {
+	status = "okay";
+};
```
Comment (erwango) [inline tests/drivers/dma/chan_blen_transfer/boards/nucleo_l152re.overlay:9]: This overlay is no more required

File: tests/drivers/dma/loop_transfer/boards/nucleo_l152re.overlay [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (c) 2021 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&dma1 {
+	status = "okay";
+};
```
Comment (erwango) [inline tests/drivers/dma/loop_transfer/boards/nucleo_l152re.overlay:9]: This overlay is no more required

**Review by erwango — CHANGES_REQUESTED** [review]

File: tests/drivers/dma/chan_blen_transfer/boards/nucleo_l152re.overlay (hunk 2 above)
Comment (FRASTM) [reply tests/drivers/dma/chan_blen_transfer/boards/nucleo_l152re.overlay:9]: removed

File: tests/drivers/dma/loop_transfer/boards/nucleo_l152re.overlay (hunk 3 above)
Comment (FRASTM) [reply tests/drivers/dma/loop_transfer/boards/nucleo_l152re.overlay:9]: removed

File: boards/arm/nucleo_f429zi/nucleo_f429zi.yaml (hunk 1 above)
Comment (FRASTM) [reply boards/arm/nucleo_f429zi/nucleo_f429zi.yaml]: unchanged

**Review by mbolivar-nordic — APPROVED** [review]
no objections

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by erwango — APPROVED** [review]
Requires one last rebase otherwise looks fine

**Comment by FRASTM** [comment]
rebase on 6c41bf6135fc8e48590689568e88b97d6228780f

**Pull request merged**